[APPSEC-87] Modify rule S5334: Educational content (Common) (#1176)
SonarSource/rspec: rules/S5334/impact.adoc

=== What is the potential impact?

In the context of a web application vulnerable to code injection: +

> After detecting the injection, attackers inject data into the vulnerable text
> field to execute malicious instructions.

Reviewer: ```suggestion
After detecting the injection point, attackers inject data into the vulnerable parameter to execute malicious instructions.
```

The impact depends on the access control measures taken on the target system
OS. In the worst-case scenario, the process that executes the code runs with
root privileges, and therefore any OS commands or programs may be affected.

Below are some real-world scenarios that illustrate the various effects of an
attacker exploiting the vulnerability.

==== Denial of service and data leaks

In this scenario, the attack aims to disrupt the organization's activities and
profit from data leaks.

The attackers:

* download the internal server's data, most likely to sell it
* modify data, send malware

> * stop services or exhaust resources (with fork bombs for example)

Reviewer: ```suggestion
* stop services or exhaust resources (e.g., fork bombs)
```

Author: I am not in favor of this change. When we defined the [styling guide](https://github.com/SonarSource/rspec/blob/master/docs/styling_guide.adoc) we decided against using abbreviations like `e.g.`. Reading the styling guide again, this is not too clear though. I think I might have to update the guide in that regard.

Optimally, the text should also not use the brackets according to the styling guide. I left them in because the text is copied from the OS command injection rule and I did not want to deviate too much.

This threat is particularly insidious if the attacked organization does not
maintain a disaster recovery plan (DRP).

> ==== Root privilege escalation and pivot

Reviewer: ```suggestion
==== Privilege escalation and pivoting
```

Author: I am in favor of leaving this as it is. Both versions are fine to me by themselves but we also use this title in the OS command injection rule and I think it would be better not to have 2 slightly different titles for the same thing.
